nixos/modules/config/users-groups.nix

```
@@ -139,6 +139,12 @@ let
         description = "The user's home directory.";
       };
 
+      homeMode = mkOption {
+        type = types.strMatching "[0-7]{1,5}";
```

i think chmod only takes four digits

```suggestion
        type = types.strMatching "[0-7]{1,4}";
```

---

According to the `chmod` man page there is a case where a fifth digit is required:
```
For directories chmod preserves set-user-ID and set-group-ID  bits  un‐
less  you  explicitly specify otherwise.  You can set or clear the bits
with symbolic modes like u+s and g-s.  To clear these bits for directo‐
ries  with  a  numeric  mode  requires  an additional leading zero like
 00755, leading minus like -6000, or leading equals like =755.
```

---

As per manual you have to use the digit 0 instead of the minus sign. I've not found a full documentation of the perl function, but I did some tests and seems to behave as the `chmod` application.

---

then you should also adjust the matching regex, because atm it only takes [0-7]

---

That accepts the example in the man "00755". Are you suggesting to make the leading character `0?` ?

---

As per the doc string of `homeMode` at the moment only the numeric format is supported.

---

tbh i am not quite sure if `[0-7]{1,5}` actually makes sense. imho it should be `[0-7]{4}` and we add a leading zero as otherwise if i set `homeMode = "7";` it would result in the owner having no rights in his directory (which doesn't really make sense to me).
so the default would be "0700" and in the perl code would need to be changed [like the last line in this toy example](https://gist.github.com/cheriimoya/968c3806106b95814e66d1a0ec57b641)

what do you think?

---

I agree that `homeMode = "7"` is uncommon, but it's exactly the same behavior as the unix `chmod` command and what most would expect. The current `[0-7]{1,5}` allows for arbitrary settings with expected results. I don't like having hardcoded settings if avoiding it is simple and gives the expected result in conformance with the standard tool for the job.

If you really don't want to allow a single digit, then I would rather propose `[0-7]{3,5}`. This way one is forced to set the basic permissions and can still use the special ones if needed. 

My preference is still for the initial proposal for the given reasons.

---

after sleeping over this, i think we can leave it like this.
IF a user is touching this option they probably know what they're doing so i guess its fine to leave it as is.
i've also tested perls chown behavior and it seems like perl is implying a leading zero, so it actually doesn't matter if you use `444`, `0444` or `00444` (only tested on directories).

---

Thanks.